### PR TITLE
diagnostic package unit tests

### PR DESCRIPTION
**Pull request by NehaNaithani**
created test for ImageDownloadManagerTest

### CODE REVIEW

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+     * When DockerUtil.getInstance().getContainer(microserviceUuid) returns null
+     * then createImageSnapshot returns
+     */
+    @Test
+    public void createImageSnapshotWithNullContainer() {
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(null);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
+    }
```
Comment (baghbidi) [inline iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: Also verify it logs `Image snapshot: container not running.`

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        error = new ArrayList<>();
+        value = new ArrayList<>();
+        value.add("path");
+        resultSetWithPath = new CommandShellResultSet<>(value, error);
+
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
```
Comment (baghbidi) [inline iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: Also, test it calls `Orchestrator#sendFileToController()` with `image-snapshot` and a File object

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        error = new ArrayList<>();
+        value = new ArrayList<>();
+        error.add("error");
+        resultSetWithPath = new CommandShellResultSet<>(value, error);
+
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
```
Comment (baghbidi) [inline iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: Test it logs the error message.

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    public void createImageSnapshotWithCommandExecuteReturnsEmpty() {
+        error = new ArrayList<>();
+        value = new ArrayList<>();
+        resultSetWithPath = new CommandShellResultSet<>(value, error);
+
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
```
Comment (baghbidi) [inline iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: Also, test it does not call `Orchestrator#sendFileToController()`

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        error = new ArrayList<>();
+        value = new ArrayList<>();
+        value.add("");
+        resultSetWithPath = new CommandShellResultSet<>(value, error);
+
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
```
Comment (baghbidi) [inline iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: Same as the above.

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        doThrow(new Exception("Error")).when(orchestrator).sendFileToController(any(), any());
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
+        verify(orchestrator, atLeastOnce()).sendFileToController(any(), any());
+
```
Comment (baghbidi) [inline iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: Delete these blank lines. Also, test it logs the error message.

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java (hunk 2 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: verify(orchestrator, atLeastOnce()).sendFileToController(any(), any());

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java (hunk 1 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java (hunk 2 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java (hunk 3 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java (hunk 4 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java (hunk 5 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java]: updated

**Pull request merged**